apps/hosts/views.py

```diff
@@ -197,13 +195,24 @@ def challenge_host_detail(request, challenge_host_team_pk, pk):
             response_data = serializer.data
             return Response(response_data, status=status.HTTP_200_OK)
         else:
-            return Response(
-                serializer.errors, status=status.HTTP_400_BAD_REQUEST
-            )
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
     elif request.method == "DELETE":
-        challenge_host.delete()
-        return Response(status=status.HTTP_204_NO_CONTENT)
+        if challenge_host_team.created_by == request.user:
+
+            if (challenge_host.user == request.user):  # when the user tries to remove himself
+                response_data = {
+                    "error": "You are not allowed to remove yourself since you are admin. Please delete the team if you want to do so!"
```

Comment:
change this `You are not allowed to remove yourself since you are admin. Please delete the team if you want to do so!` to `You are not allowed to remove yourself since you are the team admin. Please delete the team if you want to do so!`